interp/run.go

```diff
@@ -1527,6 +1534,19 @@ func callBin(n *node) {
 				break
 			}
 
+			// defType is the target type for an eventual interface wrapper.
+			var defType reflect.Type
+			if variadic >= 0 && i+rcvrOffset >= variadic {
+				defType = funcType.In(variadic)
+			} else {
+				defType = funcType.In(rcvrOffset + i)
+			}
```

Comment:
as an optimization, this could be in an else after the if of 1544.

Reply:
I tested this change and tests fails at _test/issue-1166.go because `defType` is now nil when entering the switch at run.go:1551 (was not before).

Reply:
One way that works is:
```
			// defType is the target type for a potential interface wrapper.
			var defType reflect.Type
			if getMapType != nil {
				if rt := getMapType(c.typ); rt != nil {
					defType = rt
				}
			}
			if defType == nil {
				if variadic >= 0 && i+rcvrOffset >= variadic {
					defType = funcType.In(variadic)
				} else {
					defType = funcType.In(rcvrOffset + i)
				}
			}
```

I haven't found an elegant way to invert the first condition yet though (in order to remove the `if defType == nil {`).